### ISSUE TITLE
setting LLOYD in input

### ISSUE THREAD

**Issue by rudrab** [ase2sprkkr/ase2sprkkr#3]
Hi,
I am trying to set "LLOYD" in input with:
`  calculator.input_parameters.MODE.MODE = 'lloyd'
`
But I am getting error:
`ValueError: Value 'LLOYD' for paramater MODE of type AnyOf(NREL,SREL,SP-SREL) is not valid: Required one of [NREL|SREL|SP-SREL]
`
Is there any way to fix it?
I am using the  2.2.0 version of ase2sprkkr (installed using pip -pre)

**Comment by lokik** [comment]
Hi,
LLOYD is currently a separate option, you can set it using
input_parameters.MODE.LLOYD = true
I'm not SPR-KKR expert, so I can be wrong, but I assume, that LLOYD can appear in the input file together with
any of
NREL|SREL|SP-SREL
keywords, so it is correct, if it is a separate keyword. If I'm wrong, please, let me know and
I will make an appropriate change (I will be glad if you supply a short explanation so I can
write an appropriate help message to the option).

You can also use the set_dangerous method, e.g.
input_parameters.MODE.MODE.set_dangerous('LLOYD')
to change any option to something, that is not allowed, as a temporary workaround.

Sincerelly,
Matyáš

**Comment by rudrab** [comment]
Hi,
Thanks a lot for your reply.
LLOYD can appear in the input independent of NREL|SREL|SP-SREL. 

Also, there is some ambiguity in MODE section generation. For example, to set SREL and LLOYD, I have to define:
 ```
calculator.input_parameters.MODE.MODE = "SREL"
 calculator.input_parameters.MODE.LLOYD = True
```
In my humble opinion, if it takes the same way, like:
```
 calculator.input_parameters.MODE.SREL = True
 calculator.input_parameters.MODE.LLOYD = True
```
it will be much better.

The same problem appears if we set mode in calculators:
 ```
calculator = SPRKKR(atoms=base,
                      input_parameters="SCF",
                      options={
                          "NE": 500,
                          "MODE.LLOYD": True,
                          "VXC": "PBE"
                      },
                      mpi=4)
```
works, but
```
 calculator = SPRKKR(atoms=base,
                      input_parameters="SCF",
                      options={
                          "NE": 500,
                          "SREL": True,
                          "VXC": "PBE"
                      },
                      mpi=4)
```
does not. 
Actually, I haven't find any way to define SREL/NREL inside calculator `options`.

Since both of them come under "`MODE`" section, it will be beneficial (IMHO) if they can be defined in the same way.

Please let me know if I can be of some help.

**Comment by lokik** [comment]
Hi,
the reason, why the SREL etc.... are given in another way is, that SREL is (at least I hope, please correct me if I'm wrong) not independent on e.g. NREL: only one of these options is reasonable to be present in the output file. Thus, they are not independent boolean variables, but they are given as the MODE switch, where you can choose one from the options:
it prevents you to accidentally "switch on two modes at once", which, I believe, results in silent neglecting of one of them (or in error? in fact I did not try it).

The goal of ase2sprkkr is not to give an user 1:1 interface to SPRKKR as he would write directly the input files for SPRKKR, but to give as much as fail-proof interface, as possible - in my humble is better to input a given option in a bit different way than it is done directly in the SPR-KKR input files, if the chosen way is more logical and so fail-proof.
Please, feel free to criticize the approach, if you feel that it is not the way to go. However, I think that chosen approach is reasonable here, it allows you e.g. to write:
`calcuator.input_parameters.MODE.help()` 
and see that these "options" are disjunct.


However, there were bug that prevents from setting the option value using SECTION.NAME syntax in some circumstances. I fixed the bug and uploaded a new version 2.2.1 to pip and github.  You should be able to set 

` calculator = SPRKKR(atoms=base,
                      input_parameters="SCF",
                      options={
                          "MODE": SREL,
                      },
                      mpi=4)`
                
or

` calculator = SPRKKR(atoms=base,
                      input_parameters="SCF",
                      options={
                          "MODE.MODE": SREL,
                      },
                      mpi=4)`

**Comment by rudrab** [comment]
Hi,
Thanks for the explanation and for updating the repo on Friday afternoon. 
Its working as intended now.

Regards,
